### DISCUSSION TITLE
Option to create a document by cloning template or other page.

### DISCUSSION THREAD

**Discussion by llibdude** [tinacms/tinacms#2675] [Ideas]
One of my use cases for Tina is to use it as a page builder that anyone, technical or not, could use to create pages for a website.  Speed and ease of use are important to me.

To that end, it would be nice if a user could create a new page based off a template previously created.  As an example, a template for me might be
```
<hero>
<section1>
<section2>
<section3>
```

The main benefit to templates is it gives a good starting point to a user so they don't have to generate a page from scratch, they just go in an modify already existing blocks/fields (and perhaps add/remove based on specific needs).  There could also be multiple templates based on the specific use case, which helps enforce consistency.

**Potential Solution**
I'll leave implementation up to the pros but here are my ideas around it anyway.
- In addition to the ability to create a new page, you can create a new `template`
- Templates are built exactly the same as normal pages
- They are saved in a `_template` folder inside the specified collection folder
- When creating a new page, you can optionally select a template to clone from. In this case i assume the backend would just copy the template to collection folder with the given page name.

**Easy(er?) alternative - Clone an existing page**
My end goal is to reduce clicks/confusion/intimidation of making a page from scratch.  If a user says `Hey this existing page looks a lot like a page I want to make, I want to use it as a starting point.`, then they should be able to create a new page using the other page's .mdx as a starting off point.  This of course is just templates with a slightly different workflow

**Comment by genox** [comment]
Copying/cloning an existing document would be a nice start, especially to facilitate translations for more complex content structures.
